x-pack/plugins/ml/public/application/util/url_state.ts

```diff
@@ -58,54 +58,87 @@ export function getUrlState(search: string): Dictionary<any> {
-        urlState[accessor] = {};
-      }
-
-      if (typeof attribute === 'string') {
-        if (isEqual(getNestedProperty(urlState, `${accessor}.${attribute}`), value)) {
-          return;
+        if (!Object.prototype.hasOwnProperty.call(urlState, accessor)) {
+          urlState[accessor] = {};
         }
 
-        urlState[accessor][attribute] = value;
-      } else {
-        const attributes = attribute;
-        Object.keys(attributes).forEach((a) => {
-          urlState[accessor][a] = attributes[a];
-        });
-      }
+        if (typeof attribute === 'string') {
+          if (isEqual(getNestedProperty(urlState, `${accessor}.${attribute}`), value)) {
+            return prevSearchString;
+          }
 
-      try {
-        const oldLocationSearch = stringify(parsedQueryString, { sort: false, encode: false });
+          urlState[accessor][attribute] = value;
+        } else {
+          const attributes = attribute;
+          Object.keys(attributes).forEach((a) => {
+            urlState[accessor][a] = attributes[a];
+          });
+        }
 
-        Object.keys(urlState).forEach((a) => {
-          if (isRisonSerializationRequired(a)) {
-            parsedQueryString[a] = encode(urlState[a]);
-          } else {
-            parsedQueryString[a] = urlState[a];
-          }
-        });
-        const newLocationSearch = stringify(parsedQueryString, { sort: false, encode: false });
+        try {
+          const oldLocationSearchString = stringify(parsedQueryString, {
+            sort: false,
+            encode: false,
+          });
 
-        if (oldLocationSearch !== newLocationSearch) {
-          history.push({
-            search: stringify(parsedQueryString, { sort: false }),
+          Object.keys(urlState).forEach((a) => {
+            if (isRisonSerializationRequired(a)) {
+              parsedQueryString[a] = encode(urlState[a]);
+            } else {
+              parsedQueryString[a] = urlState[a];
+            }
+          });
+          const newLocationSearchString = stringify(parsedQueryString, {
+            sort: false,
+            encode: false,
           });
+
+          if (oldLocationSearchString !== newLocationSearchString) {
+            return stringify(parsedQueryString, { sort: false });
+          }
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error('Could not save url state', error);
         }
-      } catch (error) {
-        // eslint-disable-next-line no-console
-        console.error('Could not save url state', error);
-      }
+
+        // as a fallback and to satisfy the hooks callback requirements
+        // return the previous state if we didn't need or were not able to update.
+        return prevSearchString;
+      });
     },
-    [search]
+    [searchString]
   );
 
-  return [getUrlState(search)[accessor], setUrlState];
+  const urlState = useMemo(() => parseUrlState(searchString)[accessor], [searchString]);
```

Comment:
I think it worth to mention in a comment for this hook that it does not react on the `accessor` parameter update

Reply:
Good point, the `accessor` should never be changed, I updated the code to enforce that so it only ever considers the value passed on first call: https://github.com/elastic/kibana/pull/70576/commits/3500f11ba7c08ed5c9af8b2f58e738ed1ed3d2d6#diff-42f8ed66c3f1b64589d16f1b4c9d1aa2R63-R65